### ISSUE TITLE
Boolean named function parameters

### ISSUE THREAD

**Issue by tterranigma** [niieani/bash-oo-framework#31]
I have a function that accepts a "boolean" as an argument. I want to make this argument named. I have:

```bash
import util/namedParameters
set -u

function q {
  [boolean] debug
  if [ "$debug" = "true" ]; then
    set -x
  else
    set +x
  fi
}
q true
```
Now, due to `set -u` the script fails as there is an unbound varible `__primitive_extension_fingerprint__boolean`. This variable is defined in `util/types`.

- If I remove `set -u` the `$debug` variable has a value of `:false`.
- If I copy and paste `__primitive_extension_fingerprint__boolean` from `utils/types` into `utils/namedParameters`, the script succeeds. But the variable's value has a long alphanumeric prefix.

In both cases, my check `[ "$debug" = "true"] fails. How should I check for the value of the boolean?

**Comment by niieani** [comment]
Booleans are special types, so that's why you can't compare them with strings. To tell the truth, I'm not sure if they're worth the hassle. The UUID implementation is there to make the booleans unique, i.e. you can't compare them with anything else, which also makes them a bit safer.

This is probably one of those cases where it might make sense to re-think how I've originally implemented them. Perhaps another solution would be to simply add some global variables that could be used for comparisons ($someBoolean = $true or $someBoolean = $false). The thing is, globals are a bit dangerous since you can easily override them by accident.

Going back to usage, see this part of the readme: https://github.com/niieani/bash-oo-framework#standard-library

TL;DR, you can access and assert the truthiness of the boolean like this:

```bash
function q {
  [boolean] debug
  if $var:debug; then
    set -x
  else
    set +x
  fi
}

q true
```

or like this:

```bash
function q {
  [boolean] debug
  if var: debug; then
    set -x
  else
    set +x
  fi
}

q true
```

Hope that helps!

**Comment by tterranigma** [comment]
Yes, this helped understand what `$var` and `var:` are doing. I create #33. 

There I set the boolean fingerprint to empty instead of leaving it unbound (the fingerprint is declared in `util/type`). This way the `namedParameters.sh` file can still be used standalone, which as I read was one of the original goals as well.

I find the variable approach to booleans interesting and much more straightforward and self documenting for the user. The variables can be made readonly.

Closing the issue as the question was answered!
